telethon/utils.py

```diff
@@ -617,7 +618,21 @@ def _get_metadata(file):
         else:
             stream = file
             close_stream = False
+            if getattr(file, 'seekable', None):
```

Comment:
I followed your example in the [upload_file](https://github.com/LonamiWebs/Telethon/blob/02b8f1d0077872a67b7f462b6861800c0318fc29/telethon/client/uploads.py#L557-L581) method and added rewinding the position for seekable streams and reading in non-seekable streams. What are your thoughts on this?

Reply:
I don't think smart automatic metadata extraction should read the entire file in-memory. Further, if it's read here, how can it be read again later when we can't seek?

I think we should bail out if it's not seekable. There's also the issue where it might be an "aiofile" and we need to `await`. But I'm inclined to just ignore that issue until/if it arises. You can add  comment though, `# Note: aiofiles are intentionally left out for simplicity`.